Packs/PerplexityAI/Integrations/PerplexityAI/PerplexityAI.py

```diff
@@ -0,0 +1,191 @@
+                md += f"{indent} {key}\n"
+                md += DictMarkdown(val, indent)
+            elif isinstance(val, list):
+                md += f"{indent} {key}\n"
+                md += DictMarkdown(val, indent)
+            else:
+                md += f"{indent} {key}: {val}\n"
+    elif isinstance(nested, list):
+        for val in nested:
+            md += f"{indent} []\n"
+            if isinstance(val, dict):
+                md += DictMarkdown(val, indent)
+            elif isinstance(val, list):
+                md += f"{indent} {val}\n"
+                md += DictMarkdown(val, indent)
+            else:
+                md += f"  {indent} {val}\n"
+
+    return md
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client, args: dict) -> str:
+    try:
+        client.test_module(args)
+    except DemistoException as e:
+        if 'Forbidden' in str(e):
+            return 'Authorization Error: ensure API Key is correctly set'
+        else:
+            raise e
+    return 'ok'
+
+
+def deepresearch_chat_command(client: Client, args: dict, thinking: str, citations: str, jsonout: str) -> CommandResults:
+    response = client.chat(args)
+    content = response['choices'][0]['message']['content']
+    if thinking == "no":
+        content = clean_think(content)
+    final_results = {'content': content}
+    if citations == "yes":
+        final_results['citations'] = response['citations']
+    if jsonout == "no":
+        return CommandResults(
+            readable_output=DictMarkdown(final_results, ""),
+        )
+    else:
+        return CommandResults(
+            outputs_prefix='DeepResearch',
+            outputs=final_results,
+            readable_output=DictMarkdown(final_results, "")
+        )
+
+
+def main() -> None:
+    params = demisto.params()
+    args = demisto.args()
+    command = demisto.command()
+    demisto.debug(f'Command being called is {command}')
+
+    try:
+        headers = {
+            'accept': "application/json",
+            'Authorization': f"Bearer {params.get('apikey')['password']}",
+            'Content-Type': "application/json"
+        }
+
+        # Support for Cloudflare authentication
+        cf_auth = params.get('cf_auth', None)
+        cf_client_id = None if cf_auth is None else cf_auth['identifier']
+        cf_client_key = None if cf_auth is None else cf_auth['password']
+
+        if cf_client_id is not None and cf_client_key is not None:
+            headers.update({
+                'CF-Access-Client-Id': cf_client_id,
+                'CF-Access-Client-Secret': cf_client_key
+            })
+
+        client = Client(
+            base_url=params.get('url'),
+            verify=not params.get('insecure', False),
+            headers=headers,
+            proxy=params.get('proxy', False)
+        )
+
+        if command == 'test-module':
+            result = test_module(client, {
+                'model': "sonar-deep-research",
```

Comment:
```suggestion
                'model': params.get('model', 'sonar-deep-research')
```
So you will test the model from the parameters configuration